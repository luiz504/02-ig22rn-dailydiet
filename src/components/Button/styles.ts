import { TouchableHighlight } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled(TouchableHighlight)`
  padding: 14px 24px;
  border-radius: 6px;

  justify-content: center;
  align-items: center;
  border-width: 1.5px;
  border-style: solid;
`

export const Content = styled.View`
  flex-direction: row;
  align-items: center;
  gap: 8px;
`
