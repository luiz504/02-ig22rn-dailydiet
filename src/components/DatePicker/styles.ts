import { Pressable } from 'react-native'
import styled from 'styled-components/native'
import { Text } from '../Text'

export const Button = styled(Pressable).attrs(({ theme }) => ({
  android_ripple: { color: theme.colors['black-06%'] },
}))`
  border-radius: 6px;
  border-width: 1.5px;
  border-color: ${({ theme }) => theme.colors['gray-300']};
  padding: 14px;
`
export const Label = styled(Text)``
