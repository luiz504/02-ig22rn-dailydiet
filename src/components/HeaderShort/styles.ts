import { TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled.View`
  flex-direction: row;
  padding: 17px 24px 32px 16px;
  align-items: center;
  justify-content: center;
  position: relative;
`

export const BtnReturn = styled(TouchableOpacity)`
  position: absolute;
  left: 24px;
  bottom: 28px;
`
