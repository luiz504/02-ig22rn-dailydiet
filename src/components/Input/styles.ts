import { css, styled } from 'styled-components/native'
import { TextInput as TextInputNative } from 'react-native'
type TextInputProps = {
  isFocused: boolean
}

export const StyledInput = styled(TextInputNative)<TextInputProps>`
  padding: 11px 14px;
  border-radius: 6px;
  border-width: 1.5px;

  font-size: 16px;

  ${({ isFocused, theme }) => css`
    border-color: ${isFocused
      ? theme.colors['gray-600']
      : theme.colors['gray-300']};

    color: ${theme.colors['gray-800']};
    font-family: ${theme.fontFamily.regular};
  `}
`
