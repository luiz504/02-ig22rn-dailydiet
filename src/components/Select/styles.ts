import styled, { css, DefaultTheme } from 'styled-components/native'

import { Text } from '../Text'
import { Pressable } from 'react-native'

export type Variant = 'green' | 'red'

type ContainerProps = {
  variant: Variant
  isSelected: boolean
}

const buttonsVariantStyle: Record<
  Variant,
  (theme: DefaultTheme) => ReturnType<typeof css>
> = {
  green: (theme) => css`
    border-color: ${theme.colors['green-900']};
    background-color: ${theme.colors['green-100']};
  `,
  red: (theme) => css`
    border-color: ${theme.colors['red-900']};
    background-color: ${theme.colors['red-100']};
  `,
}

export const Container = styled(Pressable)<ContainerProps>`
  flex-direction: row;
  align-items: center;
  justify-content: center;

  padding: 16px;
  gap: 8px;
  border-radius: 8px;
  border-width: 1.5px;

  border-color: ${({ theme }) => theme.colors['gray-200']};
  background-color: ${({ theme }) => theme.colors['gray-200']};

  ${({ variant, theme, isSelected }) =>
    isSelected && buttonsVariantStyle[variant](theme)}
`

type DotProps = {
  variant: Variant
}

export const Dot = styled.View<DotProps>`
  width: 8px;
  height: 8px;
  border-radius: 99999px;

  background-color: ${({ theme, variant }) =>
    variant === 'green' ? theme.colors['green-900'] : theme.colors['red-900']};
`

export const Label = styled(Text).attrs({ size: 'sm', weight: 'bold' })``
