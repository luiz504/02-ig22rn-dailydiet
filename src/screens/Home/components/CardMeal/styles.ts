import { TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled(TouchableOpacity)`
  padding: 14px 16px 14px 12px;
  border-radius: 6px;
  gap: 12px;
  border-width: 1.5px;
  border-color: ${({ theme }) => theme.colors['gray-300']};

  flex-direction: row;
  align-items: center;
`
export const HR = styled.View`
  background-color: ${({ theme }) => theme.colors['gray-400']};
  width: 1px;
  height: 100%;
`
export const StatusIndicator = styled.View`
  height: 14px;
  width: 14px;
  border-radius: 9999px;
`
