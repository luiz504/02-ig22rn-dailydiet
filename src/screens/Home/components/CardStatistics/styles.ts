import { Pressable } from 'react-native'
import styled from 'styled-components/native'
import ArrowUp from '~/assets/icons/arrow-right-up.svg'
import Animated from 'react-native-reanimated'

export const AnimatedWrapper = styled(Animated.View)`
  margin-top: 12px;
  margin-bottom: 40px;
`

export const Container = styled(Pressable)`
  padding: 20px 16px;
  border-radius: 8px;

  position: relative;
`
export const ContentAnimated = styled(Animated.View)`
  justify-content: center;
  align-items: center;
`

export const ArrowIcon = styled(ArrowUp)`
  position: absolute;
  top: 0px;
  right: 0px;
`
