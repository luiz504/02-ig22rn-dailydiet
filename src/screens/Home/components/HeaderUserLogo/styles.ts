import styled from 'styled-components/native'

export const Container = styled.View`
  padding: 24px 0;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`
export const ProfileWrapper = styled.View`
  height: 40px;
  width: 40px;
  border-radius: 20px;
  border: 2px solid ${({ theme }) => theme.colors['gray-700']};
`
