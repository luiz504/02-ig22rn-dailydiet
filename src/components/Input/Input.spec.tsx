import { fireEvent, render, screen, splitBorderColor } from '~/utils/test-utils'
import { Input } from '.'
import { theme } from '~/styles'

describe('Input Component', () => {
  it('should render correctly', () => {
    render(<Input testID="input" />)
    const inputElement = screen.getByTestId('input')

    expect(inputElement.props.cursorColor).toBe(theme.colors['gray-800'])
    expect(inputElement.props.style).toMatchSnapshot('input-not-focused')

    // Act
    fireEvent(inputElement, 'focus')

    expect(inputElement).toHaveStyle({
      ...splitBorderColor(theme.colors['gray-600']),
    })

    // Act

    fireEvent(inputElement, 'blur')
    expect(inputElement).toHaveStyle({
      ...splitBorderColor(theme.colors['gray-300']),
    })
  })
})
