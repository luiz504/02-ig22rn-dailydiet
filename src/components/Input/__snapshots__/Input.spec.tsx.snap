// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly: input-not-focused 1`] = `
{
  "borderBottomColor": "#DDDEDF",
  "borderBottomLeftRadius": 6,
  "borderBottomRightRadius": 6,
  "borderBottomWidth": 1.5,
  "borderLeftColor": "#DDDEDF",
  "borderLeftWidth": 1.5,
  "borderRightColor": "#DDDEDF",
  "borderRightWidth": 1.5,
  "borderTopColor": "#DDDEDF",
  "borderTopLeftRadius": 6,
  "borderTopRightRadius": 6,
  "borderTopWidth": 1.5,
  "color": "#1B1D1E",
  "fontFamily": "Nunito-Regular",
  "fontSize": 16,
  "paddingBottom": 11,
  "paddingLeft": 14,
  "paddingRight": 14,
  "paddingTop": 11,
}
`;
